mobile/lib/utils/selection_handlers.dart

```diff
@@ -85,3 +90,60 @@ Future<void> handleFavoriteAssets(
     }
   }
 }
+
+Future<void> handleEditDateTime(
+  WidgetRef ref,
+  BuildContext context,
+  List<Asset> selection,
+) async {
+  DateTime? initialDate;
+  String? timeZone;
+  Duration? offset;
+  if (selection.length == 1) {
+    final asset = selection.first;
+    final assetWithExif = await ref.watch(assetServiceProvider).loadExif(asset);
+    final (dt, oft) = assetWithExif.getTZAdjustedTimeAndOffset();
+    initialDate = dt;
+    offset = oft;
+    timeZone = assetWithExif.exifInfo?.timeZone;
+  }
+  final dateTime = await showDateTimePicker(
+    context: context,
+    initialDateTime: initialDate,
+    initialTZ: timeZone,
+    initialTZOffset: offset,
+  );
+  if (dateTime == null) {
+    return;
+  }
+
+  ref.watch(assetServiceProvider).changeDateTime(selection.toList(), dateTime);
```

Comment:
watch -> read in any non-build function (e.g. callback like here)

Reply:
Updated